packages/desktop-client/src/components/FinancesApp.tsx

```diff
@@ -100,6 +100,29 @@ export function FinancesApp() {
     }, 100);
   }, []);
 
+  useEffect(() => {
+    async function run() {
+      await global.Actual.waitForUpdateReadyForDownload();
```

Comment:
I haven't run this to check but doesn't this mean this will only run in an open budget and only on mount. So we're only checking once that the update is ready for download and only in the budget file. 

I think this also needs to run when the user is in the ManagementApp.tsx (the budget manager screen).

You could move it into a composable or maybe move it out into app.tsx?

It may also be worthwhile doing the setInterval thing so the user is notifed when the update is available without having to refresh the page. Some people leave tabs open for a very long time and never close their browsers.

Reply:
The notification actually comes from the serviceworker in the background, so I don't think `setInterval` is necessary here. Good point that this should be lifted up, though

Reply:
In particular, it's a little weird but the way I implemented it the promise inside this effect won't resolve until the serviceworker notifies the app of a new update

Reply:
Ah I see now. So the promise is left hanging and resolves when the service worker notifies of update. Gotcha